src/Microsoft.AspNetCore.Authentication/ClaimsTransformationMiddleware.cs

```diff
@@ -35,17 +36,22 @@ public class ClaimsTransformationMiddleware
 
         public async Task Invoke(HttpContext context)
         {
-            var handler = new ClaimsTransformationHandler(Options.Transformer, context);
+            var transform = Options.Transformer;
+            if (transform == null && Options.TransformerType != null)
+            {
+                transform = context.RequestServices.GetRequiredService(Options.TransformerType) as IClaimsTransformer;
```

Comment:
The type on options things is a bit weird (and new). MVC sorta does something similar but I need to think about this pattern a bit more before it spreads like wildfire 😄